File: E-Commerce Application/ECommerce.Domain/Product.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace ECommerce.Domain
{
    public class Product
    {
        //Id, Name, Description, Price, Stock, CategoryId

        public int ProductId { get; set; }
        public string PName { get; set; }
        public string Description { get; set; }

        public int Stock { get; set; }
        public int CategoryId { get; set; }
        public Category? Category { get; set; }



    }
}

File: ECommerceFrontend/ECommerceApp/src/app/models/product.ts
export interface Product {

    productId: number;
    pName: string;
    description: string;
    price: number;
    quantity: number;
    category?: Category;
    // imageUrl: string;

}

export interface Category {
    id: number;
    name: string;
    description: string;
   
}